File: app/src/main/java/com/app/customfontsexample/MainActivity.java
package com.app.customfontsexample;

import android.graphics.Typeface;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.core.content.res.ResourcesCompat;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        final TextView textView = findViewById(R.id.text_view_1);
        final Typeface typeface = ResourcesCompat.getFont(this, R.font.architects_daughter);

//        Typeface t = Typeface.create(typeface, 200, false);

        textView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                textView.setTypeface(typeface);
            }
        });
    }
}

File: app/src/main/java/com/app/customfontsexample/MainActivity.kt
package com.app.customfontsexample

import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.core.content.res.ResourcesCompat

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)
        val textView = findViewById<TextView>(R.id.text_view_1)
        val typeface = ResourcesCompat.getFont(this, R.font.architects_daughter)
        //        Typeface t = Typeface.create(typeface, 200, false);
        textView.setOnClickListener { textView.typeface = typeface }
    }
}